### PR TITLE
APIGW NG fix last tests

### PR DESCRIPTION
**Pull request by bentsku**
## Motivation
Fix the last following tests in order to be able to enable a new CI step to test the next gen APIGW invocation logic. 

```
FAILED tests/aws/scenario/note_taking/test_note_taking.py::TestNoteTakingScenario::test_notes_rest_api
FAILED tests/aws/services/stepfunctions/v2/services/test_apigetway_task_service.py::TestTaskApiGateway::test_invoke_base
FAILED tests/aws/services/stepfunctions/v2/services/test_apigetway_task_service.py::TestTaskApiGateway::test_invoke_with_body_post[None]
FAILED tests/aws/services/stepfunctions/v2/services/test_apigetway_task_service.py::TestTaskApiGateway::test_invoke_with_body_post[]
FAILED tests/aws/services/stepfunctions/v2/services/test_apigetway_task_service.py::TestTaskApiGateway::test_invoke_with_body_post[HelloWorld]
FAILED tests/aws/services/stepfunctions/v2/services/test_apigetway_task_service.py::TestTaskApiGateway::test_invoke_with_body_post[request_body3]
FAILED tests/aws/services/stepfunctions/v2/services/test_apigetway_task_service.py::TestTaskApiGateway::test_invoke_with_query_parameters
FAILED tests/aws/test_multiregion.py::TestMultiRegion::test_multi_region_api_gateway
```

## Changes
- fix the tests, only a few changes needed

## Testing
Successful run with the flag enabled: https://app.circleci.com/jobs/github/localstack/localstack/227232

## TODO

What's left to do:

- [x] remove the forced flag (sorry @thrau you're asked for review because of the forced flag change in config, will remove once the base PR for this is merged 😅)

### CODE REVIEW

File: localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_response.py [hunk 1]
```diff
@@ -110,19 +110,18 @@ def __call__(
                 headers[f"x-amzn-remapped-{header}"] = value
                 headers.remove(header)
 
-        # Those headers are passed through from the response headers, there might be more?
-        passthrough_headers = ("connection", "content-length")
+        # # Those headers are passed through from the response headers, there might be more?
```
Comment (cloutierMat) [inline localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_response.py:113]: Can probably remove that ;)
```suggestion
        # Those headers are passed through from the response headers, there might be more?
```

File: localstack-core/localstack/services/apigateway/helpers.py [hunk 2]
```diff
@@ -874,6 +874,9 @@ def connect_api_gateway_to_sqs(gateway_name, stage_name, queue_arn, path, accoun
                     "uri": "arn:aws:apigateway:%s:sqs:path/%s/%s"
                     % (sqs_region, sqs_account, queue_name),
                     "requestTemplates": {"application/json": template},
+                    "requestParameters": {
+                        "integration.request.header.Content-Type": "'application/x-www-form-urlencoded'"
+                    },
```
Comment (cloutierMat) [inline localstack-core/localstack/services/apigateway/helpers.py:879]: We seem to have a few "testing only" helpers in this file. Should we make a plan to move them to the tests files?

**Review by cloutierMat — APPROVED** [review]
:tada: :balloon: :tada: All the test! :eyes: Congrats!

I only left a comment about the testing configuration function in the helper file :confused:, but nothing blocking, this can easily be tackled in a follow up

File: localstack-core/localstack/services/apigateway/helpers.py (hunk 2 above)
Comment (bentsku) [reply localstack-core/localstack/services/apigateway/helpers.py:879]: Yes please 😭 We really should, I believe once we migrate the current implementation into a `legacy` folder, we can find all imports to `legacy` and refactor all of it. This function for example makes no sense and is not even that right...

**Pull request merged**